helix-core/src/main/java/org/apache/helix/HelixAdmin.java

```diff
@@ -51,6 +51,16 @@
   InstanceConfig getInstanceConfig(String clusterName, String instanceName);
 
   /**
+   * Set the instance config of an existing instance under the given cluster.
+   * @param clusterName the name of the cluster to which this instance belongs.
+   * @param instanceName the name of this instance.
+   * @param instanceConfig the new {@link InstanceConfig} that will replace the current one
+   *                       associated with this instance.
+   * @return true if the operation was successful; false otherwise.
+   */
+  boolean setInstanceConfig(String clusterName, String instanceName, InstanceConfig instanceConfig);
```

Comment:
Could do that, but doesn't `setInstanceConfig(clusterName, instanceConfig)` feel a bit odd as an API?